src/Vault.sol

```diff
@@ -227,24 +233,25 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP, Multi
         (overallCollateral, adjustedCollateral, ) = _calculateVaultCollateral(vaultId, 0, false);
     }
 
-    /// @notice Get global time-weighted stabilisation fee per USD (multiplied by DENOMINATOR)
-    /// @return uint256 Global stabilisation fee per USD (multiplied by DENOMINATOR)
-    function globalStabilisationFeePerUSDD() public view returns (uint256) {
-        return
-            globalStabilisationFeePerUSDSnapshotD +
-            (stabilisationFeeRateD * (block.timestamp - globalStabilisationFeePerUSDSnapshotTimestamp)) /
-            YEAR;
-    }
-
     /// @inheritdoc ICDP
-    function getOverallDebt(uint256 vaultId) public view returns (uint256) {
-        uint256 currentDebt = vaultDebt[vaultId];
-        uint256 deltaGlobalStabilisationFeeD = globalStabilisationFeePerUSDD() -
-            globalStabilisationFeePerUSDVaultSnapshotD[vaultId];
-        return
-            currentDebt +
-            stabilisationFeeVaultSnapshot[vaultId] +
-            FullMath.mulDiv(currentDebt, deltaGlobalStabilisationFeeD, DENOMINATOR);
+    function getOverallDebt(uint256 vaultId) public view returns (uint256 overallDebt) {
+        uint256 currentNormalizedDebt = vaultNormalizedDebt[vaultId];
+        uint256 updateTimestamp = globalFeesRateUpdateTimestamp;
+        uint256 globalFeesRate = globalFeesRateDebtAccumulatorD;
+
+        if (block.timestamp > updateTimestamp) {
+            globalFeesRate += FullMath.mulDiv(
+                (stabilisationFeeRateD * (block.timestamp - updateTimestamp)) / YEAR,
+                globalFeesRate,
+                DEBT_DENOMINATOR
+            );
+        }
+
+        overallDebt = FullMath.mulDiv(vaultNormalizedDebt[vaultId], globalFeesRate, DEBT_DENOMINATOR);
+        uint256 mintsToBePaid = vaultMintsToBePaid[vaultId];
+        if (mintsToBePaid > overallDebt) {
```

Comment:
When this becomes possible?

Reply:
look at test/AbstractVault.t.sol:testBurnDebtSuccess

Reply:
I think we can round up the normalized debt during mint, then this won't be needed:
```
vaultNormalizedDebt[vaultId] += FullMath.mulDiv(amount, DEBT_DENOMINATOR, currentNormalizationRate) + 1;
```